osbs/api.py

```diff
@@ -671,22 +672,37 @@ def create_orchestrator_build(self, **kwargs):
             raise
 
     @osbsapi
-    def get_build_logs(self, build_id, follow=False, build_json=None, wait_if_missing=False):
+    def get_build_logs(self, build_id, follow=False, build_json=None, wait_if_missing=False,
+                       decode=False):
         """
         provide logs from build
 
-        NOTE: since there is no way to determine the text encoding in use
-        the bytes are returned exactly as they came from the container.
+        NOTE: there's no way to determine the text encoding in use, so if asked
+        to decode, we always assume utf-8 encoding, otherwise, we return the
+        bytes exactly ast they came from the container.
 
         :param build_id: str
         :param follow: bool, fetch logs as they come?
         :param build_json: dict, to save one get-build query
         :param wait_if_missing: bool, if build doesn't exist, wait
-        :return: None, bytes, or iterable of bytes
+        :param decode: bool, whether or not to decode logs as utf-8
+        :return: None, iterable of bytes or iterable of strs
         """
-        return self.os.logs(build_id, follow=follow, build_json=build_json,
+        logs = self.os.logs(build_id, follow=follow, build_json=build_json,
                             wait_if_missing=wait_if_missing)
 
+        if isinstance(logs, GeneratorType):
```

Comment:
One nice side-effect I'm seeing from the way I'm doing things is that the data returned stays consistent -- it's always an iterable generator, and the content matches what we're expecting. I've tried a few different variations, and they result in sometimes returning a string instead of an iterable with the string inside it, or the string gets hacked up oddly like I see with the variant you suggested. It's not the cleanest thing in the world, but the added complexity inside get_build_logs makes the test cases cleaner, as well as making follow-on work (get_orchestrator_build_logs) cleaner, since they know they'll always get a generator (or None) back.